src/engine/runtime.js

```diff
@@ -439,6 +439,11 @@ Runtime.prototype.startHats = function (requestedHatOpcode,
     }
     var instance = this;
     var newThreads = [];
+
+    for (var opts in optMatchFields) {
+        optMatchFields[opts] = optMatchFields[opts].toUpperCase();
+    }
```

Comment:
It feels a little icky to me to overwrite the arguments coming in. Would you please update to just compare `hatFields[matchField].value.toUpperCase() !== optMatchFields[opts].toUpperCase()`?